### PR TITLE
Added support for CollectSourceInformation flag in runsettings

### PR DESCRIPTION
**Pull request by Shyam-Gupta**
1. If CollectSourceInformation is already present in the run settings then it will not get updated
2. If it is not present then it will default to value of DesignMode flag and will be added to run settings
3. Added automated tests

### CODE REVIEW

File: src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs [hunk 1]
```diff
@@ -362,6 +385,8 @@ public bool ResultsDirectorySet
             Justification = "XmlDocument.XmlResolver is not available in core. Suppress until fxcop issue is fixed.")]
         public override XmlElement ToXml()
         {
+            System.Diagnostics.Debugger.Launch();
```
Comment (Faizan2304) [inline src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs]: Please remove this.

File: src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs [hunk 2]
```diff
@@ -85,6 +90,7 @@ public RunConfiguration() : base(Constants.RunConfigurationSettingsName)
             this.disableAppDomain = false;
             this.disableParallelization = false;
             this.designMode = false;
+            this.shouldCollectSourceInformation = this.designMode;
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs]: nit : Replace `this.shouldCollectSourceInformation = this.designMode;` with `this.shouldCollectSourceInformation = false;`

File: src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs [hunk 3]
```diff
@@ -87,18 +89,36 @@ public static void UpdateRunSettingsWithUserProvidedSwitches(XPathNavigator runS
...
+
+        /// <summary>
+        /// Adds node under RunConfiguration setting. Noop if node is already present.
+        /// </summary>
+        private static void AddNodeIfNotPresent<T>(XPathNavigator runSettingsNavigator, string nodePath, string nodeName, T nodeValue)
+        {
             // Navigator should be at Root of runsettings xml, attempt to move to /RunSettings/RunConfiguration
             if (!runSettingsNavigator.MoveToChild(RunSettingsNodeName, string.Empty) ||
-!runSettingsNavigator.MoveToChild(RunConfigurationNodeName, string.Empty))
+                !runSettingsNavigator.MoveToChild(RunConfigurationNodeName, string.Empty))
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs]: nit : remote whitespaces.

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 4]
```diff
@@ -287,13 +277,33 @@ private bool TryUpdateDesignMode(string runsettingsXml, out string updatedRunSet
...
+
                 updatedRunSettingsXml = navigator.OuterXml;
             }
+            
 
-            return true;
+            return settingsUpdated;
         }
-
+
```
Comment (harshjain2) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: nit : remote whitespaces

**Comment by harshjain2** [comment]
@dotnet-bot test this please.

File: src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs (hunk 3 above)
Comment (Shyam-Gupta) [inline src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs]: Its required for indentation.

File: src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs (hunk 2 above)
Comment (Shyam-Gupta) [inline src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs]: Done.

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs (hunk 4 above)
Comment (Shyam-Gupta) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Done

**Review by harshjain2 — APPROVED** [review]

**Pull request merged**